arbnode/espresso_key_manager.go

```diff
@@ -0,0 +1,166 @@
+package arbnode
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/solgen/go/mocksgen"
+	"github.com/offchainlabs/nitro/util/signature"
+)
+
+type EspressoKeyManagerInterface interface {
+	HasRegistered() (bool, error)
+	Register(signFunc func([]byte) ([]byte, error)) error
+	GetCurrentKey() *ecdsa.PublicKey
+	SignHotShotPayload(message []byte) ([]byte, error)
+	SignBatch(message []byte) ([]byte, error)
+}
+
+var _ EspressoKeyManagerInterface = &EspressoKeyManager{}
+
+type EspressoTEEVerifierInterface interface {
+	RegisterSigner(opts *bind.TransactOpts, attestation []byte, addr []byte, teeType uint8) error
+	RegisteredSigners(signer common.Address, teeType uint8) (bool, error)
+}
+
+type EspressoTEEVerifier struct {
+	contract *mocksgen.EspressoTEEVerifierMock
+	l1Client *ethclient.Client
+}
+
+func NewEspressoTEEVerifier(contract *mocksgen.EspressoTEEVerifierMock, l1Client *ethclient.Client) *EspressoTEEVerifier {
+	return &EspressoTEEVerifier{contract: contract, l1Client: l1Client}
+}
+
+func (e *EspressoTEEVerifier) RegisterSigner(opts *bind.TransactOpts, attestation []byte, addr []byte, teeType uint8) error {
+	tx, err := e.contract.RegisterSigner(opts, attestation, addr, teeType)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Waiting for register signer tx to be mined", "tx", tx.Hash())
+
+	receipt, err := bind.WaitMined(context.Background(), e.l1Client, tx)
+	if err != nil {
+		return err
+	}
+
+	if receipt.Status != types.ReceiptStatusSuccessful {
+		return errors.New("transaction failed")
+	}
+
+	log.Info("Register signer tx succeeded", "tx", tx.Hash())
```

Comment:
nit: Can we also print out which signer was registered? Will be helpful for debugging later